torch/csrc/utils/python_numbers.h

```diff
@@ -85,9 +86,9 @@ inline int64_t THPUtils_unpackIndex(PyObject* obj) {
 
 inline bool THPUtils_checkDouble(PyObject* obj) {
 #if PY_MAJOR_VERSION == 2
-  return PyFloat_Check(obj) || PyLong_Check(obj) || PyInt_Check(obj);
+  return PyFloat_Check(obj) || PyLong_Check(obj) || PyInt_Check(obj) || torch::utils::is_numpy_scalar(obj);
```

Comment:
don't you want to guard the call to is_numpy_scalar here?  Because this will throw an exception if we didn't compile with numpy and the object isn't a PyFloat, PyLong, PyInt?